addons/point_of_sale/models/pos_order.py

```diff
@@ -995,12 +992,8 @@ def _compute_amount_line_all(self):
     @api.onchange('product_id')
     def _onchange_product_id(self):
         if self.product_id:
-            if not self.order_id.pricelist_id:
-                raise UserError(
-                    _('You have to select a pricelist in the sale form !\n'
-                      'Please set one before choosing a product.'))
-            price = self.order_id.pricelist_id._get_product_price(
-                self.product_id, self.qty or 1.0)
+            pricelist = self.order_id.pricelist_id or self.env['product.pricelist']
```

Comment:
From my little tests, each POS is linked to journals in a single currency (I know, I told you otherwise the other day, and I was wrong), but this might need confirmation

Reply:
Yes, but what if the journal currency is not the (pricelist/product) currency ?